cpp/arduino/avr/iomxx0_1.h

```diff
@@ -351,6 +351,17 @@
    Last two letters:   EEAR address.  */
 #define __EEPROM_REG_LOCATIONS__ 1F2021
 
+/* According to https://ww1.microchip.com/downloads/en/devicedoc/atmel-2549-8-bit-avr-microcontroller-atmega640-1280-1281-2560-2561_datasheet.pdf,
+ * the Atmel ATmega640/V-1280/V-1281/V-2560/V-2561/V microcontrollers each have 
+ * 4Kbytes of EEPROM and since this file is included from things that seem
+ * to match that description, it would be helpful to know that EEPROM is 
+ * available and know its size. This macro is defined for many other models,
+ * so we will add it here.
+ * 
+ * See https://github.com/Arduino-CI/arduino_ci/issues/166#issuecomment-703904394.
+ */
+#define EEPROM_SIZE     (4096)

```

Comment:
It might be more appropriate to conditionally set this variable based on the board being used, which can be done in the default configuration here:
https://github.com/Arduino-CI/arduino_ci/blob/master/misc/default.yml#L102

Reply:
I can certainly try that. By way of background, where did those files come from? Why does EEPROM_SIZE exist in over 40 of them?

Reply:
I copied them from the Arduino project itself, `arduino-1.8.5/hardware/tools/avr/avr/include/avr` to get things to compile... some hand-edits were made.  I have very little information on how/why Arduino structures these files.

Reply:
>where did those files come from?

It's avr-libc:
https://www.nongnu.org/avr-libc/
Nothing Arduino-specific. This is just the standard open source AVR toolchain.

Reply:
>Why does EEPROM_SIZE exist in over 40 of them?

Because there are a lot of different AVR microcontrollers.

Reply:
I'd really love to find a way to avoid packaging all of these files in this repo but I'm at a loss for how to do so (while enabling a non-AVR compiler to supply all the `#define` statements some alternate way).

Reply:
> Because there are a lot of different AVR microcontrollers.

But why isn't it defined for Mega2560? That microcontroller has EPROM in it? I was hoping that the macro would be a reliable way to determine whether the feature is present. Instead it seems that it has `__EEPROM_REG_LOCATIONS__` but not the size.

Reply:
The files are mostly out of the way and don't bother me much; I just find myself expecting more of them than I should given my limited knowledge!

Reply:
I notice that, while the [megaAVR EEPROM library uses `EEPROM_SIZE`](https://github.com/arduino/ArduinoCore-megaavr/blob/1.8.3/libraries/EEPROM/src/EEPROM.h#L127), [the AVR EEPROM library uses `E2END`](https://github.com/arduino/ArduinoCore-avr/blob/1.8.3/libraries/EEPROM/src/EEPROM.h#L127).

Reply:
I see now. The macros are artifacts of the implementation, not a way of giving information to clients of the library, and different implementations will have different macros. That makes sense and is reasonable.

Further investigation shows that E2END exists in 230 files and many times is (0x0). Perhaps it is a more reliable macro to use.